templates/terraform/schema_property.erb

```diff
@@ -103,9 +103,29 @@
   <% end -%>
 	<% end -%>
 <% elsif property.is_a?(Api::Type::NameValues) -%>
+  <% if property.key_type == 'Api::Type::String' && property.value_type == 'Api::Type::String' -%>
   Elem: &schema.Schema{Type: schema.TypeString},
-  <% if not (property.key_type == 'Api::Type::String' and property.value_type == 'Api::Type::String') -%>
+  <% elsif string_to_object_map?(property) -%>
+  Elem: &schema.Resource{
+    Schema: map[string]*schema.Schema{
+      "<%= property.key_name -%>": {
+        Type:     schema.TypeString,
+        Required: true,
+        <% if force_new?(property, object) -%>
+        ForceNew: true,
+        <% end -%>
+      },
+      <% order_properties(property.value_type.properties).each do |prop| -%>
+        <%= lines(build_schema_property(prop, object)) -%>
+      <% end -%>
+    },
+  },
+  <% if !property.set_hash_func.nil? -%>
+  Set: <%= property.set_hash_func -%>,
```

Comment:
Should this hash function always hash the map's 'key'?

Reply:
I can confirm for sure tomorrow, but I believe when I was testing this out that that wouldn't work, because if an item in the object changed that wasn't the key, the hash wouldn't change and so whatever does the diffing would stop looking.

Reply:
Yup, confirmed.